src/util/open-browser.js

```diff
@@ -1,190 +1,20 @@
-  const executable = launcher[browser].create(params);
-
-  ProcessLauncher.call(executable, require('child_process').spawn, require('./temp-dir'), window);
-
-  executable.start();
-
-  app.on('window-all-closed', function() {
-    executable.kill();
-  });
-}
-
-var ProcessLauncher = function(spawn, tempDir, timer) {
-  var self = this;
-  var onExitCallback;
-  var killTimeout = 2000;
-
-  this._tempDir = tempDir.getPath('/karma-' + this.id.toString());
-
-  this.start = function() {
-    var url = '';
-    tempDir.create(self._tempDir);
-    self._start(url)
-  };
-
-  this.kill = function() {
-    if (!self._process) {
-      return;
-    }
-
-    onExitCallback = done;
-    self._process.kill();
-    self._killTimer = timer.setTimeout(self._onKillTimeout, killTimeout)
-  };
-
-  this._start = function(url) {
-    self._execCommand(self._getCommand(), self._getOptions(url))
-  };
-
-  this._getCommand = function() {
-    return env[self.ENV_CMD] || self.DEFAULT_CMD[process.platform]
-  };
-
-  // Normalize the command, remove quotes (spawn does not like them).
-  this._normalizeCommand = function(cmd) {
-    if (cmd.charAt(0) === cmd.charAt(cmd.length - 1) && '\'`"'.indexOf(cmd.charAt(0)) !== -1) {
-      cmd = cmd.substring(1, cmd.length - 1)
-    }
-
-    return path.normalize(cmd)
+  const opts = {
+    browser: browser,
+    proxy: "localhost:1338"
   };
 
-  this._execCommand = function(cmd, args) {
-    if (!cmd) {
-      // disable restarting
-      self._retryLimit = -1;
-
-      return self._clearTempDirAndReportDone('no binary')
-    }
-
-    cmd = this._normalizeCommand(cmd);
-
-    self._process = spawn(cmd, args);
-
-    var errorOutput = '';
-
-
-    // Node 0.8 does not emit the error
-    if (process.versions.node.indexOf('0.8') === 0) {
-      self._process.stderr.on('data', function(data) {
-        var msg = data.toString();
-
-        if (msg.indexOf('No such file or directory') !== -1) {
-          self._retryLimit = -1;
-          errorOutput = 'Can not find the binary ' + cmd + '\n\t' +
-          'Please set env variable ' + self.ENV_CMD
-        } else {
-          errorOutput += msg
-        }
-      })
+  browserLauncher(function(err, launch) {
+    if (err) {
+      return console.error(err);
     }
-  };
-
-  this._onProcessExit = function(code, errorOutput) {
-
-    var error = null;
 
-    if (self.state === self.STATE_BEING_CAPTURED) {
-      error = 'cannot start'
-    }
-
-    if (self.state === self.STATE_CAPTURED) {
-      error = 'crashed'
-    }
-
-    self._process = null;
-    if (self._killTimer) {
-      timer.clearTimeout(self._killTimer);
-      self._killTimer = null
-    }
-    self._clearTempDirAndReportDone(error)
-  };
-
-  this._clearTempDirAndReportDone = function(error) {
-    tempDir.remove(self._tempDir, function() {
-      self._done(error)
-      if (onExitCallback) {
-        onExitCallback()
-        onExitCallback = null
+    launch('http://www.uxebu.com/', opts, function(err) {
```

Comment:
We could open a page like `http://james.awesome` and use a default mapping to point this request to a local file. This would even better demonstrate the power of james I think